calyx-py/calyx/builder.py

```diff
@@ -221,7 +221,7 @@ def case(
         width = self.infer_width(signal)
         ifs = []
         for branch, controllable in cases.items():
-            std_eq = self.eq(width, f"{signal.name}_eq_{branch}", signed)
+            std_eq = self.eq(width, self.generate_name(f"{signal.name}_eq_{branch}"), signed)
```

Comment:
> the same two things need to be compared repeatedly?

Admittedly, I'm having a hard time parsing this. Which two things are compared repeatedly?

The reason for this tweak was to allow the user to have multiple case statements depending on the same signal; their bodies can be totally different though. This seems like a reasonable use-case to me?

For example, in `binheap/round_robin.py` we have two case statements depending on `flow_in.out` (i.e. the flow of our incoming packet):
```
comp.case(flow_in.out, binheap_invokes)
```
so that we push to the heap with the correct rank and
```
comp.case(flow_in.out, rank_ptr_incrs)
```
so that we update our state correctly by incrementing the right rank pointer.

Previously, the eDSL would complain both calls to `comp.case` instantiate comparators called `flow_in_out_eq_<branch number>`.